google/cloud/forseti/services/inventory/base/resources.py

```diff
@@ -225,27 +230,60 @@ def fetch(cls, client, resource_key):
 
     @cached('iam_policy')
     def getIamPolicy(self, client=None):
-        return client.get_project_iam_policy(self['projectId'])
+        if self.enumerable():
+            return client.get_project_iam_policy(self['projectId'])
+        return {}
 
     @cached('billing_info')
     def getBillingInfo(self, client=None):
-        return client.get_project_billing_info(self['projectId'])
+        if self.enumerable():
+            return client.get_project_billing_info(self['projectId'])
+        return {}
+
+    @cached('enabled_apis')
+    def getEnabledAPIs(self, client=None):
+        enabled_apis = []
+        if self.enumerable():
+            enabled_apis = client.get_enabled_apis(self['projectId'])
+
+        self._enabled_service_names = frozenset(
+            (api.get('serviceName') for api in enabled_apis))
+        return enabled_apis
 
     def key(self):
         return self['projectId']
 
     def enumerable(self):
-        return self['lifecycleState'] not in ['DELETE_REQUESTED']
+        return self['lifecycleState'] == 'ACTIVE'
 
     def billing_enabled(self):
         return self.getBillingInfo().get('billingEnabled', False)
 
-    @cached('compute_api_enabled')
-    def compute_api_enabled(self, client=None):
-        if not self.billing_enabled():
-            # Compute API depends on billing being enabled
-            return False
-        return client.is_compute_api_enabled(projectid=self['projectId'])
+    def is_api_enabled(self, service_name):
```

Comment:
Nice work on logging enabled APIs with parent so we don't have to raise warnings when API is not enabled in target resource. LGTM